sourcegraph/sourcegraph
# Code Insight: Move repositories methods out of context API

## Background

In this PR, I removed a few repositories-related API methods and put them as close as possible to their consumers. We don't need to have them in the global context-based API instance anymore. 

In the future, a minimal number of methods in the API class will help us conduct a global refactoring described in [this doc](https://docs.google.com/document/d/1GYLfV6VCB42YyoqWCoM9ACFaEnPQz1n5_veArEVCus4/edit#) about code insight FE types and API.   


## Test plan
- Check that repo autosuggest in the creation UI pages works as expected (provides repo suggestions)
- Check that 1-click creation insight flow works as expected (resolves repo:filters and converts repo filter value into list of repositories for the repo field in the creation UI)



## App preview:

- [Web](https://sg-web-vk-clean-up-api-backend-1.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
